### PR TITLE
[Impeller] Reland move more files to core.

### PR DESCRIPTION
**Pull request by dnfield**
Relands https://github.com/flutter/engine/pull/40801
See https://github.com/flutter/engine/pull/40811
Includes changes in https://github.com/flutter/engine/pull/40810

This got reverted due to an unhappy merge missing an include in a modified file. Fixed that up and it should build now.

### CODE REVIEW

**Review by bdero — APPROVED** [review]

**Review by zanderso — COMMENTED** [review]
There was an unanswered question from @bdero on the original PR. Was that discussed out-of-band?

File: impeller/core/resource_binder.cc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/core/resource_binder.h"
+
+namespace impeller {
+
+//
```
Comment (zanderso) [inline impeller/core/resource_binder.cc:9]: What's this?

Comment (dnfield) [reply impeller/core/resource_binder.cc:9]: This is how all the other otherwise header-only TUs are handled in impeller

**Comment by dnfield** [comment]
Due to github magic I missed that question. The answer (I'll add it there too), is that `ResourceBinder`'s real purpose is to make sure that the generated code can depend exclusively on core and not on renderer, because renderer has shaders in it now.

If we add some other command type that's radically different from `ComputeCommand` and `Command`, we can add a different type that's like `ResourceBinder` but more suited to that new (currently hypothetical) command.

**Pull request merged**